TestGoFuzzCrashers fails on 32-bit systems

On i386, the tests fail by trying to create too large a slice:
```
=== RUN   TestGoFuzzCrashers
--- FAIL: TestGoFuzzCrashers (0.00s)
panic: runtime error: makeslice: len out of range [recovered]
	panic: runtime error: makeslice: len out of range

goroutine 374 [running]:
panic(0x824b780, 0x186a60a8)
	/usr/lib/go-1.7/src/runtime/panic.go:500 +0x331
testing.tRunner.func1(0x186a8200)
	/usr/lib/go-1.7/src/testing/testing.go:579 +0x14f
panic(0x824b780, 0x186a60a8)
	/usr/lib/go-1.7/src/runtime/panic.go:458 +0x40b
github.com/streadway/amqp.readLongstr(0x8345dc0, 0x1885a0a0, 0x0, 0x0, 0x0, 0x0)
	/build/golang-github-streadway-amqp-0.0~git20170313.0.afe8eee/obj-i686-linux-gnu/src/github.com/streadway/amqp/read.go:112 +0xeb
github.com/streadway/amqp.readTable(0x8345dc0, 0x1885a0a0, 0x0, 0x0, 0x0)
	/build/golang-github-streadway-amqp-0.0~git20170313.0.afe8eee/obj-i686-linux-gnu/src/github.com/streadway/amqp/read.go:269 +0x7e
github.com/streadway/amqp.(*reader).parseHeaderFrame(0x18824f1c, 0x18851610, 0xefbfbd5b, 0x0, 0x0, 0x0, 0x0)
	/build/golang-github-streadway-amqp-0.0~git20170313.0.afe8eee/obj-i686-linux-gnu/src/github.com/streadway/amqp/read.go:363 +0x3cc
github.com/streadway/amqp.(*reader).ReadFrame(0x18824f1c, 0x0, 0x0, 0x0, 0x0)
	/build/golang-github-streadway-amqp-0.0~git20170313.0.afe8eee/obj-i686-linux-gnu/src/github.com/streadway/amqp/read.go:64 +0x2f0
github.com/streadway/amqp.TestGoFuzzCrashers(0x186a8200)
	/build/golang-github-streadway-amqp-0.0~git20170313.0.afe8eee/obj-i686-linux-gnu/src/github.com/streadway/amqp/read_test.go:21 +0x20b
testing.tRunner(0x186a8200, 0x829c998)
	/usr/lib/go-1.7/src/testing/testing.go:610 +0x8c
created by testing.(*T).Run
	/usr/lib/go-1.7/src/testing/testing.go:646 +0x2a5
exit status 2
FAIL	github.com/streadway/amqp	1.985s
```

AFAICS a slice can not be larger than the maximal value of (signed) `int`. Since the test is trying to create a slice of length 4290768831, it fails in this case since this value exceeds the range of a 32-bit signed integer.

--- a/read.go
+++ b/read.go
@@ -109,6 +109,11 @@ func readLongstr(r io.Reader) (v string, err error) {
 		return
 	}
 
+	// slices can't be longer than max int32 value
+	if length > (^uint32(0) >> 1) {
+		return
+	}
+
 	bytes := make([]byte, length)
 	if _, err = io.ReadFull(r, bytes); err != nil {
 		return

